pkg/runtime/types.go

```diff
@@ -120,26 +130,324 @@ type Unknown struct {
+		return str
+	}
+	return ""
+}
+
+func getNestedMap(obj map[string]interface{}, fields ...string) map[string]string {
+	if m, ok := getNestedField(obj, fields...).(map[string]interface{}); ok {
+		strMap := make(map[string]string, len(m))
+		for k, v := range m {
+			if str, ok := v.(string); ok {
+				strMap[k] = str
+			}
+		}
+		return strMap
+	}
+	return nil
+}
+
+func setNestedField(obj map[string]interface{}, value interface{}, fields ...string) {
+	m := obj
+	if len(fields) > 1 {
+		for _, field := range fields[0 : len(fields)-1] {
+			if _, ok := m[field].(map[string]interface{}); !ok {
+				m[field] = make(map[string]interface{})
+			}
+			m = m[field].(map[string]interface{})
+		}
+	}
+	m[fields[len(fields)-1]] = value
+}
+
+func setNestedMap(obj map[string]interface{}, value map[string]string, fields ...string) {
+	m := make(map[string]interface{}, len(value))
+	for k, v := range value {
+		m[k] = v
+	}
+	setNestedField(obj, m, fields...)
+}
+
+func (u *Unstructured) setNestedField(value interface{}, fields ...string) {
+	if u.Object == nil {
+		u.Object = make(map[string]interface{})
+	}
+	setNestedField(u.Object, value, fields...)
+}
+
+func (u *Unstructured) setNestedMap(value map[string]string, fields ...string) {
+	if u.Object == nil {
+		u.Object = make(map[string]interface{})
+	}
+	setNestedMap(u.Object, value, fields...)
+}
+
+func extractOwnerReference(src interface{}) (metatypes.OwnerReference, error) {
+	v := src.(map[string]interface{})
+	return metatypes.OwnerReference{
+		Kind:       getNestedString(v, "kind"),
+		Name:       getNestedString(v, "name"),
+		APIVersion: getNestedString(v, "apiVersion"),
+		UID:        (types.UID)(getNestedString(v, "uid")),
+	}, nil
+}
+
+func setOwnerReference(dest interface{}, src metatypes.OwnerReference) error {
+	v, ok := dest.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("dest %v has to be of type map[string]interface()", dest)
+	}
+	setNestedField(v, src.Kind, "kind")
+	setNestedField(v, src.Name, "name")
+	setNestedField(v, src.APIVersion, "apiVersion")
+	setNestedField(v, string(src.UID), "uid")
+	return nil
+}
+
+func getOwnerReferences(object map[string]interface{}) ([]map[string]interface{}, error) {
+	field := getNestedField(object, "metadata", "ownerReferences")
+	if field == nil {
+		return nil, fmt.Errorf("cannot find field metadata.ownerReferences in %v", object)
+	}
+	ownerReferences, ok := field.([]map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("expect metadata.ownerReferences to be a slice in %v", object)
+	}
+	return ownerReferences, nil
+}
+
+func (u *Unstructured) GetOwnerReferences() []metatypes.OwnerReference {
+	var ret []metatypes.OwnerReference
+	original, err := getOwnerReferences(u.Object)
+	if err != nil {
+		glog.Error(err)
+		return nil
+	}
+	ret = make([]metatypes.OwnerReference, 0, len(original))
+	for i := 0; i < len(original); i++ {
+		ref, err := extractOwnerReference(original[i])
```

Comment:
This never returns an error. Also, it's only used here. Better to just append the struct literal directly.